projects/discussion-ui/src/lib/elements/related-discussion/related-discussion.component.ts

```diff
@@ -35,26 +38,28 @@ export class RelatedDiscussionComponent implements OnInit, OnChanges {
   }
 
   fetchRelatedDiscussionData(cid: number) {
-    this.fetchSingleCategoryLoader = true
+    this.fetchSingleCategoryLoader = true;
     this.discussionService.fetchSingleCategoryDetails(cid).subscribe(
       (data: NSDiscussData.ICategoryData) => {
         this.relatedDiscussions = [];
         _.filter(data.topics, (topic) => {
-          if (this.topicId != topic.tid) {
-            this.relatedDiscussions.push(topic)
+          if (topic.deleted === 0 && this.topicId !== topic.tid) {
+            this.relatedDiscussions.push(topic);
           }
         })
-        this.fetchSingleCategoryLoader = false
+        this.fetchSingleCategoryLoader = false;
       },
       (err: any) => {
         console.log('Error in fetching category details')
         // this.openSnackbar(err.error.message.split('|')[1] || this.defaultError)
-        this.fetchSingleCategoryLoader = false
-      })
+        this.fetchSingleCategoryLoader = false;
+      });
   }
 
   getDiscussion(discuss) {
-    this.router.navigate([`${this.configService.getRouterSlug()}${CONSTANTS.ROUTES.DISCUSSION}topic/${discuss.slug}`]);
+    this.passDiscussData.emit(discuss);
```

Comment:
all components should implement navigate method, 
This methohd should call (common property for all components)
this.nvigationData.emit(discuss);